### PR TITLE
Test ESMF from conda and remainder PyPI deps

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] HISTORY.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Attempts to install xscen in a pure Python (tox) environment ~with the `install-esmf-action`~
    * (`install-esmf-action`-installed ESMF appears to cause some segfaults with pytest. I'd love to get to the bottom of it, but I know when to quit. Here's hoping this isn't exposing a problem that we might again have further on.)
* More specific packaging wildcards to prevent warnings emitted in `tox`
* Disables `tox-conda`
* `tox` now offers a means for installing `esmpy` from a tag/branch (via `ESMF_VERSION` environment variable)
* Linters are now called alphabetically
* Manifest is much more specific about what is installed
* Pin on xarray now in `setup.py`
* Re-adds a dev recipe to the `setup.py`
* Removes xdoctest dependency (wasn't being used).

### Does this PR introduce a breaking change?

No. `ESMF_VERSION` is only needed if `tox` is called with `-esmpy` flag. The `tox` config was always dependent on an environment having `esmpy`, therefore there are few instances where it would work anyway (requires `tox-current-env` or `tox-conda`).

### Other information:

https://github.com/esmf-org/install-esmf-action (I think I'll wait for the next version before trying this again).

### CODE REVIEW

**Comment by Zeitsperre** [comment]
I have no idea why I keep getting Segfaults on GitHub Actions. There has to be something strange going on with the test setup that isn't memory safe. Is this possible?

**Comment by aulemahal** [comment]
Can you remind us of what's wrong with installing ESMF through mamba on the CI ?

And why is `nc-time-axis` a dependency, I don't think we plot anything in xscen ? (I mean outside the docs)

**Comment by Zeitsperre** [comment]
I'm going to squash these commits to not pollute the history.

File: environment.yml [hunk 1]
```diff
@@ -26,7 +26,8 @@ dependencies:
   - shapely >= 2
   - sparse
   - toolz
-  - xarray
+  # FIXME: Unpin xarray when xclim 0.46 is released
+  - xarray <2023.09.0
```
Comment (RondeauG) [inline environment.yml:30]: We waited enough for this PR, so it's not an issue anymore 😆

File: setup.py [hunk 2]
```diff
@@ -59,14 +59,32 @@ def run(self):
     "shapely>=2.0",
     "sparse",
     "toolz",
-    "xarray",
+    # FIXME: Unpin xarray when xclim 0.46 is released
+    "xarray<2023.09.0",
```
Comment (RondeauG) [inline setup.py:63]: Same here.

**Review by RondeauG — COMMENTED** [review]
I don't see any major problem, but I don't think that I am able to review the details... Let me know if you just want me to approve this.

**Review by RondeauG — APPROVED** [review]

**Pull request merged**